### DISCUSSION TITLE
The axios http request is not working when i build the tauri app

### DISCUSSION THREAD

**Discussion by chemistwang** [tauri-apps/tauri#8433] [Q&A]
when i use `npm run tauri build` to build my app. I found my axios is not work, it shows ` tauri://localhost/`, but i use development mode the network is work
![image](https://github.com/tauri-apps/tauri/assets/32005560/31142e12-b05b-45d9-8134-a2825acdd937)
![image](https://github.com/tauri-apps/tauri/assets/32005560/53c61d2d-5126-4c52-9015-b56180030e3d)
![image](https://github.com/tauri-apps/tauri/assets/32005560/ef623394-26b4-488e-abad-bac4136e0f7c)

**Comment by FabianLars** [comment]
`tauri://localhost/` that's tauri's "server" (not a real server). Can you try `npm run tauri build -- --debug` (output in target/debug/ ! ) and check the devtools console again? Or did you already see any errors/warnings somewhere?

And just to be sure, your first screenshot is not part of the tauri app but a _seperate_ server, right?

**Reply by chemistwang** [reply]
Well, i was prepare to Integrate tauri into existing React project, so the first screenshot is one part of my code.
![image](https://github.com/tauri-apps/tauri/assets/32005560/c5737906-92f2-4535-8e09-75ecbe2bcd40)
The `api` folder contains axios, the `setupProxy.js` file was my first screenshot.
The screenshot below was `build mode` and `development mode` at same time.
![image](https://github.com/tauri-apps/tauri/assets/32005560/ad64ecc5-2396-4350-829b-20a4ebf65b1c)
The left devtools console shows the network errors, but the right can work. Is it possible that it is a problem with http and https?

**Reply by chemistwang** [reply]
I eliminate http/https problem because i enabled the following configurations 
![image](https://github.com/tauri-apps/tauri/assets/32005560/b7cfb80d-577e-4855-b7d3-07b0a63caad9)
![image](https://github.com/tauri-apps/tauri/assets/32005560/53f83348-9e46-45d8-bd63-bf628673cd4f)

**Reply by FabianLars** [reply]
The allowlist only matters if you use `@tauri-apps/api`. axios is not affected.

Does it show actual error messages? Maybe in the console tab?

But yeah, your setup looks like it requires a running nodejs runtime/server which doesn't exist in a tauri app so i assume that's the issue. (I'm not a web/js dev so i can't really go into detail here since i don't understand your code just from those screenshots)

**Reply by chemistwang** [reply]
There is no error messages from tauri in the console tab, just network error message coming from my code. But wow, I appreciate your thoughts that give me some ideas, I was confused some important concepts about tauri architecture.(I'm a js dev and a tauri noob ). Have I used the wrong method for network requests，I should use the `@tauri-apps/api/http` instead of the `axios`?

**Reply by FabianLars** [reply]
> I should use the @tauri-apps/api/http instead of the axios?

Not necessarily. most people use tauri's api only if they have to in case the native fetch (including axios) doesn't work because of cors restrictions for example.

**Reply by chemistwang** [reply]
Thank you. I will give a response when I find the solution.

**Reply by dondxniel** [reply]
@chemistwang were you able to find a solution? I'm experiencing the same issue.

**Reply by suhaotian** [reply]
Try [xiorjs](https://github.com/suhaotian/xior), the modern axios, for example:

```ts
import { fetch } from '@tauri-apps/plugin-http';
import axios from 'xior';

export const http = xior.create({
  baseURL: 'https://www.tauri.app',
  fetch,
});

async function test() {
  const { data } = await http.get('/');
  return data;
}
```
